src/model/Note.cpp

```diff
@@ -0,0 +1,73 @@
+#include "model/Note.hpp"
+
+namespace doma {
+namespace model {
+
+Note::Note(const NoteId &id, const std::string &name, const std::string &text)
+    : isDone_(false), id_(id), name_(name), text_(text), dateCreate_(ModelClock::now()) {}
+
+NoteId Note::getId() const {
+  return id_;
+}
+
+Timestamp Note::getDateCreate() const {
+  return dateCreate_;
+}
+
+Timestamp Note::getDateChange() const {
+  return dateChange_;
+}
+
+Timestamp Note::getDateDone() const {
+  return dateDone_;
+}
+
+std::string &Note::getName() {
+  return name_;
+}
+
+std::string &Note::getText() {
+  return text_;
+}
+
+std::vector<TagId> &Note::getTagIds() {
+  return tagIds_;
+}
+
+bool Note::isDone() const {
+  return isDone_;
+}
+
+void Note::changeName(const std::string &name) {
+  name_ = name;
+  dateChange_ = ModelClock::now();
+}
+
+void Note::changeText(const std::string &text) {
+  text_ = text;
+  dateChange_ = ModelClock::now();
+}
+
+void Note::addTag(const TagId &tagId) {
+  tagIds_.push_back(tagId);
+  dateChange_ = ModelClock::now();
+}
+
+bool Note::removeTag(const TagId &tagId) {
+  for (auto i = 0u; i < tagIds_.size(); i++) {
```

Comment:
1) Please use methods, if `std::`-libriary provides this functionality -- less our code is better
2) Please use `std::set<>` instead of `std::vector<>`. `std::set<>` cannot allow store duplicate items by design.

```
#include <iostream>
#include <set>

int main() {
  using NoteId = uint32_t;

  std::set<NoteId> notes {1, 2, 3, 4};
  const auto itNotes = notes.find(2);
  if (itNotes != notes.end()) {
    notes.erase(itNotes);
  } else {
    // notes has not the element
  }
  std::cout << notes.size() << std::endl;

  return 0;
}
```

Reply:
done